spring-cloud-gcp-autoconfigure/src/test/java/com/google/cloud/spring/autoconfigure/firestore/GcpFirestoreEmulatorAutoConfigurationTests.java

```diff
@@ -38,19 +39,25 @@
  */
 class GcpFirestoreEmulatorAutoConfigurationTests {
 
+	GcpProjectIdProvider projectId = () -> "projectId";
+
 	private final ApplicationContextRunner contextRunner = new ApplicationContextRunner()
 			.withConfiguration(AutoConfigurations.of(
-					GcpContextAutoConfiguration.class,
 					GcpFirestoreEmulatorAutoConfiguration.class,
-					GcpFirestoreAutoConfiguration.class));
+					GcpFirestoreAutoConfiguration.class))
+			.withBean("projectId", GcpProjectIdProvider.class, () -> projectId);
 
 	@Test
 	void testEmulatorEnabledConfig() {
+
+		CredentialsProvider mockedCredentialsProvider = () -> mock(NoCredentials.class);
```

Comment:
Now that you have a handle to the specific mock used for the default credentials provider, could you replace the negative assertion `assertThat(defaultCredentialsProvider).isNotInstanceOf(NoCredentialsProvider.class);` with a positive `assertThat(defaultCredentialsProvider).isSameAs(mockedCredentialsProvider);`